test/src/CMakeLists.txt

```diff
@@ -13,6 +13,7 @@ target_link_libraries(DLAF_gtest_hpx_main
   PUBLIC
     gtest
   PRIVATE
+    dlaf.core
```

Comment:
`gtest_mpihpx_main` has been changed with the new `dlaf::initialize`, but it's not the case for `gtest_hpx_main`. 
Since I see you are adding the dependency here, I think that something is missing there.

Reply:
Indeed, it should be in both. Good catch!

Reply:
What about the `gtest_mpi_main`?

As soon as we enable CUDA support, `dlaf::initialize` starts using HPX for the queue, so it implies the presence of MPI.

Are we going to drop this since DLAF has HPX as strong requirement, so there is no need to have just MPI based tests?

@rasolca ?

Reply:
MPI only test should be kept to test basic MPI functionalities (e.g. communicator management, ...).
As they do not require dlaf::init, I don't see any problem to keep them as they are now.